packstream/packstream-specification-1.md

```diff
@@ -809,15 +827,67 @@ DateTimeZoneId::Structure(
 )
 ```
 
-- The `seconds` are seconds since the adjusted [Unix epoch](https://en.wikipedia.org/wiki/Epoch_(computing)). This is not [UTC](https://en.wikipedia.org/wiki/Coordinated_Universal_Time).
-- The `tz_id` is an identifier for a specific time zone, such as `"Europe/Paris"`.
+- The `tz_id` specifies the timezone name as understood
+  by [the timezone database](https://en.wikipedia.org/wiki/Tz_database).
+- The `seconds` elapsed since the [Unix epoch](https://en.wikipedia.org/wiki/Epoch_(computing)), often referred as a
+  Unix timestamp, **added** to the offset derived from the named timezone and specified the point in time.
+- The `nanoseconds` are what remains after the last second of the datetime. The amount of nanoseconds ranges from 0 to
+  999_999_999 (`_` separator added here and later for clarity).
 
-To convert to UTC:
+For instance, the serialization of the point in time denoted as `1970-01-01T02:15:00 Europe/Paris` (and `42`
+nanoseconds) can be implemented as follows:
 
+- retrieve the offset of the named timezone for that point in time, here +1 hour, i.e. `3_600` seconds (the resolution
+  is not always defined, read the following `Known Limitations` section to learn more)
+- compute the UTC time, i.e. `1970-01-01T01:15:00`.
+- compute the difference between that UTC time and the Unix epoch, which is 1h15, i.e. `8_100` seconds.
+- add the resolved offset of 1 hour, i.e. `3_600` seconds to the above difference, which yields `11_700` (`8_100`+`3_600`)
+
+The resulting `DateTime` instance is therefore as follows:
 ```
-utc_nanoseconds = (seconds * 1000000000) + nanoseconds - get_offset_in_nanoseconds(tz_id)
+{
+  seconds: 11700
+  nanoseconds: 42,
+  tz_id: "Europe/Paris"
+}
 ```
 
+The deserialization of such a `DateTime` structure expectedly happens in reverse:
+
+- retrieve the offset of the named timezone for that point in time, here +1 hour, i.e. `3_600` seconds (the resolution
```

Comment:
This line is hand-wavy: we don't have a point in time yet, we just have seconds, nanoseconds, and a tz_id.

Reply:
I mean as a physical entity, the point in time that "exists" in the physical world

Reply:
Sure, but you have not computed it yet. We are in the process of converting a number `seconds` into such a point that we then can use in what ever language this is being implemented in. In other words: how do I get that offset while just knowing a number of seconds?

Reply:
To get to the core of the problem: this is exactly where the protocol is broken. Generally, you can't. So we should at least describe the "best effort" approach implemented in our drivers.

Reply:
I agree but I'm not sure how to describe with more precision since it is actually subject to ambiguities